core/src/main/java/org/testcontainers/containers/TestContainer.java

```diff
@@ -0,0 +1,194 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.Info;
+import lombok.NonNull;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.traits.*;
+import org.testcontainers.containers.wait.Wait;
+import org.testcontainers.containers.wait.WaitStrategy;
+import org.testcontainers.utility.SelfReference;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.concurrent.Future;
+import java.util.function.Consumer;
+
+public interface TestContainer<SELF extends TestContainer<SELF>> extends LinkableContainer, SelfReference<SELF>, TraitsSupport<SELF>,
+        // old commands like `withExposedPorts` are supported with Support interfaces
+        Command.Support<SELF>,
```

Comment:
actually, `with(new ` will auto-complete all available traits (even non-standard ones), but I think I got your point :)

In general - I don't mind until we have both ways of applying traits: pre-defined `withSomething()`-like methods and generic `with(...)` method for any other Trait.